### DISCUSSION TITLE
Injections not working?

### DISCUSSION THREAD

**Discussion by augustocdias** [nvim-treesitter/nvim-treesitter#2221] [Q&A]
Hi,

I'm trying to inject SQL language in string literals in my rust code and I tried like this:

```lua
require('vim.treesitter.query').set_query(
    'rust',
    'injections',
    [[
((
  (raw_string_literal) @constant
  (#match? @constant "(SELECT|select|INSERT|insert|UPDATE|update|DELETE|delete).*")
) @injection.content (#set! injection.language "sql"))
]]
) ```

When I tried this in the treesitter playground website it worked but it is not highlighting SQL neovim... Am I doing something wrong?

**Comment by theHamsta** [comment]
We don't have any SQL parser in our parser list. I think there was a PR but it wasn't finished. So

- the parser is missing
- the highlight queries for SQL are missing
- we have a different injection capture. Just use `@sql` on the spot to inject SQL
- you probably want to trim quotes from the raw string from the injection
